docs/developing/how-to-review-a-pull-request.rst

```diff
@@ -0,0 +1,94 @@
+============================
+How to review a pull request
+============================
+
+This is our shared, collaborative how-to for reviewing a pull request at
+Hypothesis. Like everything else in our documentation, this isn't set in stone
+but is a living, collaborative document. If you want to question or change
+something in this guide, open a GitHub issue or pull request.
+
+
+-------------------
+Aims of code review
+-------------------
+
+What we aim to achieve by code review at Hypothesis:
+
+* Foster and enact a positive, patient and friendly internal culture for our
+  team
+* Improve working relationships between developers
+* Relieve tension by giving positive feedback
+* Learn from the code that you're reviewing
+* Break down code silos, giving each member a broad exposure to different parts
+  of the codebase
+* Mentor and educate
+* Find bugs and improve code quality
+
+------
+Don'ts
+------
+
+* Don't be blunt and just point out the things that are wrong and that you want
+  changed, without saying anything else.
+
+* Don't harshly criticize code, and don't use personal tone, hyperbole, or
+  demanding, challenging, insulting, impatient or passive-aggressive language
+  when criticizing code.
+
+* Don't be a back-seat coder. Consider your suggestions carefully and pick the
+  best ones.
+
+* Don't break the rule of no surprises. Suggestions should be evidence-based
+  not opinion-based. For example, based on our
+  :doc:`documented coding standards <code-style>` rather than on conflicting
+  opinions about trade-offs.
+
+----
+Do's
+----
+
+* Praise good code
+* Ask, don't tell. Ask questions rather than making statements
+* Ask questions, don't make demands
+* Avoid accusatory *why* questions ("Why didn't you just <do what you think they should have done>?")
+* When making a suggestion, also give the reason why you think the change
+  might be an improvement
+* Use the checklist below
+* Refer to the :doc:`coding standards <code-style>`, and use them as the
+  foundation of your review
+
+---------
+Checklist
+---------
+
+This is a checklist to follow before submitting your code for review,
+and for the reviewer to use to know what to look for when reviewing:
```

Comment:
* Separate major refactorings and functional changes. For smaller refactorings, I think it is fine to include those as separate commits within a PR
* If the code contains any changes that were non-obvious when you started development (eg. a workaround for a bug in a particular library or browser), make sure this is clearly highlighted. Include links to any relevant documentation or discussion
* Unless there is a clear reason not to do so, code style should be consistent with 1) The code around the change, 2) The project in which the change is being made, 3) Applicable style guides

As an aside, it would be an interesting exercise to grab review comments from the past N months and comb through them to see what issues come up frequently.